### PR TITLE
Fix issue #26

### PR DESCRIPTION
**Pull request by ty0x2333**
I used this library in my application. But when I submitted the application to the app store for review, I received a warning email. same as #26.

I see in README that "on iOS you will have to include all permission options when you want to submit your app". This makes this library almost impossible to use on iOS.

I added some macro definitions to remove useless permissions.  (reference: https://github.com/delba/permission)

**This PR will not change any default behavior.**

By default, these macro definitions do not exist, and the code will default that they are enabled. Unless the user specifies their value in the Podfile.

### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)

feature

### :arrow_heading_down: What is the current behavior?

same as before

### :new: What is the new behavior (if this is a feature change)?

Users can remove unused permission codes on iOS.

### :boom: Does this PR introduce a breaking change?

No

### :bug: Recommendations for testing

Usage: https://github.com/Baseflow/flutter-permission-handler/commit/296982baf48be88aac22f019086c10ec727e7523

### :memo: Links to relevant issues/docs

#26

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/BaseflowIT/flutter-geolocator/blob/develop/CONTRIBUTING.md))
- [x] Relevant documentation was updated
- [x] Rebased onto current develop

### CODE REVIEW

**Comment by SpajicM** [comment]
Finally a proper fix for this! Thanks!

File: README.md [hunk 1]
```diff
@@ -53,7 +53,59 @@ For this plugin to work you will have to add permission configuration to your `A
...
 - [Info.plist](https://github.com/BaseflowIT/flutter-permission-handler/blob/develop/example/ios/Runner/Info.plist)
 
-> IMPORTANT: On iOS you will have to include all permission options when you want to submit your App. This is because the `permission_handler` plugin touches all different SDKs and because the static code analyser (run by Apple upon App submission) detects this and will assert if it cannot find a matching permission option in the `Info.plist`. More information about this can be found [here](https://github.com/BaseflowIT/flutter-permission-handler/issues/26). Unfortunately we don't have a good solution for this.    
+> IMPORTANT: ~~On iOS you will have to include all permission options when you want to submit your App.~~ This is because the `permission_handler` plugin touches all different SDKs and because the static code analyser (run by Apple upon App submission) detects this and will assert if it cannot find a matching permission option in the `Info.plist`. More information about this can be found [here](https://github.com/BaseflowIT/flutter-permission-handler/issues/26).
```
Comment (mvanbeusekom) [inline README.md]: Would be nice if you could write a small summary explaining the macros. Currently the documentation mentions you need to include all permissions in de info.plist and then it jumps directly into the "Podfile" script without explaining what it is.

**Comment by mvanbeusekom** [comment]
Resolves: #26 
Resolves: #179

**Comment by jikeyt** [comment]
@ty0x2333 I need only camera and photos permission. I added the following code in the podfile, excluded those two, cleaned and rebuilt. But once uploaded to TestFlight, I'm still getting the error. Am I missing something? PFA.

Screenshot 2020-01-09 at 5 11 40 PM

**Comment by mvanbeusekom** [comment]
@jikeyt just checking the obvious, which version of the plugin are you using? This feature is not yet released to pub and should only work if you reference the `develop` branch (or the fork from @ty0x2333).

**Comment by mvanbeusekom** [comment]
We just (5 minutes ago) released version 4.1.0 which contains the contents of this PR.

**Comment by ramoncardena** [comment]
Thanks so much for this fix @ty0x2333!

**Comment by brunozanoelo** [comment]
Hello @mvanbeusekom. I´m developin with flutter for Android and iOS, I don´t have any Podfile on my project, because I use windows. Where should I change those permission?

**Comment by mvanbeusekom** [comment]
Hi @brunozanoelo, I you want to develop for iOS you will have to compile your project on a Apple machine running macOS, there is no escaping to this. So if you stick to Windows you will not be able to develop the iOS App (or at least compile it). In this case you will only be able to develop for Android and the web (and in the near future Windows and Linux Desktop).

Most likely Flutter noticed you are running on Windows when you initially created your project and didn't setup the iOS part of the project (in your project root there should be an `ios` folder is you want to release to iOS, this is also the folder that would contain the `Podfile` file).

In your case there will not be an iOS App and thus no need to worry about the permission configuration yet. Once you get hold of a Apple machine with macOS and Xcode you will be able to generate the iOS part of the Flutter project and you will automatically also have a `Podfile`.

**Comment by brunozanoelo** [comment]
Hello @mvanbeusekom, Thank you for you reply.

I don´t have an mac computer, so I´m building this with [Codemagic](https://codemagic.io/apps), with him, I´m able to build and automatically publish into Play Store and Apple Store. However, the changes to run the app inside Podfile is very difficult for me, since I need to access the cloud machine pretty fast and change the Podfile everytime I build one version, and actually, this is not  working so far.

It seens, I´m the different one, but buying one MAC will be pretty expensive in my contry.

**Comment by SpajicM** [comment]
@brunozanoelo I've used MacInCloud to do iOS specific things, it's pretty cheap if you only need it for a couple of days, and first 24h were free.

File: README.md [hunk 2]
```diff
@@ -53,7 +53,88 @@ For this plugin to work you will have to add permission configuration to your `A
...
 
-> IMPORTANT: On iOS you will have to include all permission options when you want to submit your App. This is because the `permission_handler` plugin touches all different SDKs and because the static code analyser (run by Apple upon App submission) detects this and will assert if it cannot find a matching permission option in the `Info.plist`. More information about this can be found [here](https://github.com/Baseflow/flutter-permission-handler/issues/26). Unfortunately we don't have a good solution for this.    
+> IMPORTANT: ~~On iOS you will have to include all permission options when you want to submit your App.~~ This is because the `permission_handler` plugin touches all different SDKs and because the static code analyser (run by Apple upon App submission) detects this and will assert if it cannot find a matching permission option in the `Info.plist`. More information about this can be found [here](https://github.com/BaseflowIT/flutter-permission-handler/issues/26).
+
+On iOS, the permission_handler plugin use [macros](https://github.com/BaseflowIT/flutter-permission-handler/blob/develop/ios/Classes/PermissionHandlerEnums.h) to control whether a permission is supported.
```
Comment (noordawod) [inline README.md:58]: This URL is no longer valid. Any chance to submit a pr to correct it?

File: README.md [hunk 3]
```diff
@@ -53,7 +53,88 @@ For this plugin to work you will have to add permission configuration to your `A
...
+
+By default, all the permissions listed [here](https://github.com/Baseflow/flutter-permission-handler#list-of-available-permissions) are supported.
```
Comment (noordawod) [inline README.md:60]: This URL is no longer valid. Any chance to submit a pr to correct it?

File: README.md [hunk 4]
```diff
@@ -53,7 +53,88 @@ For this plugin to work you will have to add permission configuration to your `A
...
+> 1. Add the following to your `Podfile` file:
+>
+>    ```ruby
+>    post_install do |installer|
+>      installer.pods_project.targets.each do |target|
+>        target.build_configurations.each do |config|
+>          ... # Here are some configurations automatically generated by flutter
+>    
+>          # You can remove unused permissions here
+>          # for more infomation: https://github.com/BaseflowIT/flutter-permission-handler/blob/develop/ios/Classes/PermissionHandlerEnums.h
```
Comment (noordawod) [inline README.md:73]: This URL is no longer valid. Any chance to submit a pr to correct it?

**Pull request merged**